EventListener/IsGrantedListener.php

```diff
@@ -72,6 +74,23 @@ public function onKernelController(FilterControllerEvent $event)
         }
     }
 
+    private function getArguments(FilterControllerArgumentsEvent $event)
+    {
+        $namedArguments = $event->getRequest()->attributes->all();
+        $argumentMetadatas = $this->argumentMetadataFactory->createArgumentMetadata($event->getController());
+
+        // loop over each argument value and its name from the metadata
+        foreach ($event->getArguments() as $index => $argument) {
+            if (!isset($argumentMetadatas[$index])) {
+                throw new \LogicException(sprintf('Could not find any argument metadata for argument %d of the controller.', $index));
+            }
+
+            $namedArguments[$argumentMetadatas[$index]->getName()] = $argument;
+        }
```

Comment:
Is there any issue with variadics here? I'm assuming that if there are 3 arguments to the controller, then `$this->argumentMetadataFactory->createArgumentMetadata()` will always return 3 items exactly.

Reply:
this is indeed an issue for variadic arguments, as a variadic argument in the metadata will generate multiple values in the final array.

Reply:
a better solution might be to loop over the metadata instead (and handle the case where the metadata is a variadic argument, by building an array with this argument and all following ones, as done inside the function)